src/libstd/keyword_docs.rs

```diff
@@ -1578,12 +1578,108 @@ mod type_keyword {}
 
 #[doc(keyword = "unsafe")]
 //
-/// Code or interfaces whose [memory safety] cannot be verified by the type system.
+/// Code or interfaces whose [memory safety] cannot be verified by the type
+/// system.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+/// The `unsafe` keyword has two uses: to declare the existence of contracts the
+/// compiler can't check, and to declare that a programmer has checked that
+/// these contracts have been upheld.
+///
+/// Unsafe Rust add some capabilities to Safe Rust:
+///
+/// - Dereference [raw pointers]
+/// - Call `unsafe` functions (including foreign functions, compiler intrinsics,
+/// and the raw allocator)
+/// - Implement `unsafe` [`trait`]s
+/// - Mutate [`static`]s (including [`extern`]al ones)
+/// - Access fields of [`union`]s
+///
+/// # Dereferencing raw pointers
+///
+/// ```rust
+/// let a = 3;
+/// let b = &a as *const _;
+/// // SAFETY: we know that b will always be a valid address.
```

Comment:
b is not _always_ a valid address, b is valid _in this scope_. If we returned from this scope, `a` would be dropped and b would be a pointer to freed memory.
```suggestion
/// // SAFETY: a has not been dropped and references are always aligned, so b is a valid address.
```